src/actions/functionList.js

```diff
@@ -0,0 +1,52 @@
+
+// TODO move into a class and export methods to hide state
+let refreshTime = 5*1000
+let interval = null
+
+let intervalCounter = 0
+
+export function refreshFunctionList() {
+  return dispatch => {
+    dispatch({type:'FUNCTION_LIST_LOAD', status:'PENDING'})
+    fetch('/system/functions',
+    {
+        method: "GET", 
+        credentials: 'include'
+    })
+        .then(res => res.json())
+        .then(response => {
+            if (response) {
+              dispatch({type:'FUNCTION_LIST_LOAD', status:'SUCCESS', functionList: response})
+            }
+        }).catch(error => {
+          console.error(error)
+          dispatch({type:'FUNCTION_LIST_LOAD', status:'FAILED', message: error.message})
+        })
+  }
+}
+
+export function startPollFunctions() {
+  return dispatch => {
+
+    intervalCounter++
+    if(interval) return
```

Comment:
With this here, the `refreshFunctionList` is only ever called once. Using `clearInterval()`, the interval is never "falsy", it's only cleared so that it won't run any more.

Remove this `if` check so that the interval will be re-enabled when the Functions page is mounted again

Reply:
Clear should never be called. Once mounted, it always polls the function list. 

The reason for the debounce is because we want to make sure there's never multiple intervals running.

Reply:
Wait, crap. I did add in the stop on unmount